tensorflow/lite/micro/python/py_cc_headers.bzl

```diff
@@ -0,0 +1,59 @@
+"Repository rule for creating a repository with the C headers of a python package."
+
+load("@tflm_pip_deps//:requirements.bzl", py_bzl_package = "requirement")
+
+_destdir = "include"
+
+_build = """
+cc_library(
+    name = "cc_library",
+    hdrs = glob(["%s/**/*.h"], allow_empty=False),
+    includes = ["%s"],
+    visibility = ["@//tensorflow/lite/micro:__subpackages__"],
+)
+""" % (_destdir, _destdir)
+
+def _make_include_path(bzl_package_path, py_package, prefix):
+    p = bzl_package_path.get_child(py_package)
+    if prefix:
+        for s in prefix.split("/"):
+            p = p.get_child(s)
+    return p
+
+def _tflm_py_cc_headers_impl(ctx):
+    # Symlink from the directory containing the headers, in the repository
+    # created by @python_rules pip_install(), to _destdir within this new
+    # repository.
+    any_file = py_bzl_package(ctx.attr.py_package) + ":BUILD.bazel"
+    bzl_package_path = ctx.path(Label(any_file)).dirname
+    src = _make_include_path(bzl_package_path, ctx.attr.py_package, ctx.attr.prefix)
+    ctx.symlink(src, _destdir)
+
+    # Provide a cc_library target containing all the headers within _destdir
+    # via the top-level package's BUILD file.
+    ctx.file("BUILD", content = _build, executable = False)
+
+tflm_py_cc_headers = repository_rule(
+    implementation = _tflm_py_cc_headers_impl,
+    local = True,
+    attrs = {
+        "py_package": attr.string(mandatory = True),
+        "prefix": attr.string(mandatory = True),
+    },
+)
+"""Repository rule for creating a new repository `name` with a top-level
+Bazel package containing the C headers from the Python distribution package
+named `py_package`. The top-level Bazel package provides a target ":cc_library"
+to be used as a dependency in targets building against the headers.
+
+For example:
+
```

Comment:
let's use this location for a full illustration of the steps:

1. how to create a repo with the repository_rule defined here
2. what the directory structure in .bazel/cache/<>/ looks like
3. how this new repository is used from a BUILD target that depends on it.

Reply:
Extensive comment added. #2 covered by a comment up in the implementation, as the rule user (the audience of this particular comment) shouldn't care.